src/components/Capture/capture.js

```diff
@@ -60,23 +61,42 @@ class Capture extends Component {
     checkIfHasWebcam( hasWebcam => this.setState({hasWebcam}) )
   }
 
-  validateCapture = (id, valid) => {
-    const { actions, method } = this.props
-    actions.validateCapture({ id, valid, method})
+  maxAutomaticCaptures = 3
+
+  uploadCaptureToOnfido = () => {
+    this.setState({uploadInProgress: true})
+    const {validCaptures, method, side, token, nextStep,
+      actions: {confirmCapture}
+    } = this.props
+    const capture = validCaptures[0]
+    postToOnfido(capture, method, token, this.state.advancedValidation,
+      (apiResponse) => {
+        confirmCapture({method, id: capture.id, onfidoId: apiResponse.id})
+        this.confirmEvent(method, side)
+        nextStep()
+      },
+      this.onApiError
+    )
   }
 
-  maxAutomaticCaptures = 3
+  confirmEvent = (method, side) => {
+    if (method === 'document') {
+      if (side === 'front') events.emit('documentCapture')
+      else if (side === 'back') events.emit('documentBackCapture')
+    }
+    else if (method === 'face') events.emit('faceCapture')
+  }
 
   createCapture(payload) {
     const { actions, method, side } = this.props
     payload.side = side
     actions.createCapture({method, capture: payload, maxCaptures: this.maxAutomaticCaptures})
   }
 
-  onServerResponse = (response) => {
-    const { actions } = this.props
-    const valid = response.valid
-    this.validateCapture(response.id, valid)
+  onValidationServiceResponse = (response) => {
+    const {id, valid} = response
```

Comment:
It's a bit odd that we get the ID from the response when we send the ID in the request. I think this comes from websockets when we could not tell which request matches which response. Now we use HTTP we already know what the ID is before the response arrives.